### ISSUE TITLE
Optimize pre-commit hook

### ISSUE THREAD

**Issue by 3commascapital** [pavlovcik/uad-ui#319]
In order to keep development workflow clean and without delays, commits should be fairly thin in terms of checks. Devs should be capable of rebasing their own branch locally if they need to fix anything before pushing, which can reject after many larger checks of the codebase such as spell check and lints. It may also be worthwhile to consider dropping prettier altogether, given that it is basically doing the same job as eslint. I would propose the following checklist, not necessarily in this order.

- [x] drop prettier
- [ ] shift to pre-push rather than commit
- [x] utilize workspaces
- [ ] fix [cross package ignore requirements](https://github.com/ubiquity/ubiquity-dollar/pull/318/files#diff-e7e195d3d8fa5b82fa51ed952f3a179d429cbc46c0432ce1f7b357c54c162822R6) by scoping linting appropriately
- [x] possible to utilize npm packages instead of lib folders?

**Comment by 0x4007** [comment]
- I haven't done research on the matter regarding the capability differences of `prettier` vs `eslint` but from what I understand, they handle different types of issues. 
- I think on pre-push vs on commit makes a ton of sense for this. 
- I'm pretty sure we already use yarn workspaces.
- I can't speak on this @0xcodercrane might have some inputs.
- I'm pretty sure best-practices for Foundry is to use lib folders

**Comment by cuddle-dev** [comment]
1/ prettier is used to format codebase defined as prettier config file but eslint is used for checking code style. We need to keep both
2/ I don't have much context about the advantages of pre-push. but I think both are same at the point they run before pushing. lol. 
3/ we are already using yarn workspaces. Do you have another meaning for `workspaces` @3commascapital ?
4/ eslintrc will not need to check solidity files. Would you give us the detailed plan how you're gonna fix it? anyway if you're confident, lets give it a try.
5/ Yeah agree with @pavlovcik . we need to keep the lib folders as Foundry suggests.

**Comment by 0x4007** [comment]
> 2/ I don't have much context about the advantages of pre-push. but I think both are same at the point they run before pushing. lol.

Pre-commit runs every time the developer commits. This may happen several times before they push (however if they commit using vscode then it will be commit+pull+push all in one, every time)

Pre-push is particularly useful for developers who commit using their command line. They can commit several times and then when they push it should format one time. 

Continuing on this line of thinking, is it a bad idea to handle formatting just when we're looking to merge into development (on pull request?) Ultimately thats the only time it really matters that the code is formatted correctly I believe.

**Comment by 0x4007** [comment]
> * fix [cross package ignore requirements](https://github.com/ubiquity/ubiquity-dollar/pull/318/files#diff-e7e195d3d8fa5b82fa51ed952f3a179d429cbc46c0432ce1f7b357c54c162822R6) by scoping linting appropriately

We're still waiting on input for this. Once that is clarified we can release this bounty into the devpool!

**Comment by sergfeldman** [comment]
> > * fix [cross package ignore requirements](https://github.com/ubiquity/ubiquity-dollar/pull/318/files#diff-e7e195d3d8fa5b82fa51ed952f3a179d429cbc46c0432ce1f7b357c54c162822R6) by scoping linting appropriately
> 
> We're still waiting on input for this. Once that is clarified we can release this bounty into the devpool!

@3commascapital Can you please provide your input on this?

**Comment by 3commascapital** [comment]
yeah, so issue was created when i was iterating on the cspell update. i think this is now fixed given that i figured out how to utilize the package appropriately. can be closed
